test/controllers/configs_controller_test.rb

```diff
@@ -1,9 +1,42 @@
 require 'test_helper'
 
 class ConfigsControllerTest < ActionDispatch::IntegrationTest
-  test "should get index" do
-    get configs_index_url
-    assert_response :success
+  describe 'non-admin redirects' do
+    [:data_volunteer, :cm].each do |role|
+      it "should deny access as a #{role}" do
+        delete destroy_user_session_path
+        user = create :user, role: role
+        config = create :config
+        sign_in user
+
+        get configs_path
+        assert_redirected_to authenticated_root_path
+
+        patch config_path(config, params: { options: ['no'] })
```

Comment:
this is a good flag, they should def be identical. I'll standardize them before I put this up for final review.